app/src/main/java/org/oppia/android/app/devoptions/markchapterscompleted/MarkChaptersCompletedFragmentPresenter.kt

```diff
@@ -78,6 +78,14 @@ class MarkChaptersCompletedFragmentPresenter @Inject constructor(
               )
           }
         }
+      } else if (binding.isAllChecked == true) {
+        binding.isAllChecked = false
+        getMarkChaptersCompletedViewModel().getItemList().forEach { viewModel ->
+          if (viewModel is ChapterSummaryViewModel) {
+            if (!viewModel.checkIfChapterIsCompleted())
+              chapterUnselected(viewModel.chapterIndex, viewModel.nextStoryIndex)
+          }
+        }
```

Comment:
This branch contains code of your previous branch too. 
I suggest two solutions:
1. Either undo-the-changes in this branch by checking at the `Files Changed` tab on https://github.com/oppia/oppia-android/pull/4392
2. Or, create a new branch+PR for this issue and in this case the branch should be created from `develop`

Reply:
Update applied.